src/components/TodoModal/TodoDetailModal/useTodoDetail.ts

```diff
@@ -0,0 +1,49 @@
+import { Todo } from '@/types/interface';
+import { useEffect, useState } from 'react';
+
+function useTodoDetail(todo: Todo) {
+  const [done, setDone] = useState(todo.done);
+  const [title, setTitle] = useState(todo.title);
+  const [goal, setGoal] = useState(todo.goal);
+  const [fileUrl, setFileUrl] = useState(todo.fileUrl);
+  const [fileType, setFileType] = useState<string | null>(null);
+  const [fileName, setFileName] = useState<string | null>(null);
+  const [linkUrl, setLinkUrl] = useState(todo.linkUrl);
+  const [isModified, setIsModified] = useState(false);
```

Comment:
```suggestion
  const [state, setState] = useState({
    done: todo.done,
    title: todo.title,
    goal: todo.goal,
    fileUrl: todo.fileUrl,
    fileType: null as string | null,
    fileName: null as string | null,
    linkUrl: todo.linkUrl,
    isModified: false,
  });
```
[Suggestion] 이렇게 쓰는 것도 고려해볼 수 있을 것 같아요~